openapi2/openapi2.go

```diff
@@ -43,16 +65,49 @@ func (swagger *Swagger) AddOperation(path string, method string, operation *Oper
 	pathItem.SetOperation(method, operation)
 }
 
+func (swagger *Swagger) AddPathExtensions(path string, extensions map[string]interface{}) {
+	paths := swagger.Paths
+	if paths == nil {
+		paths = make(map[string]*PathItem, 8)
+		swagger.Paths = paths
+	}
+	pathItem := paths[path]
+	if pathItem == nil {
+		pathItem = &PathItem{}
+		paths[path] = pathItem
+	}
+	pathItem.Extensions = extensions
+}
+
 type PathItem struct {
-	Ref        string     `json:"$ref,omitempty"`
-	Delete     *Operation `json:"delete,omitempty"`
-	Get        *Operation `json:"get,omitempty"`
-	Head       *Operation `json:"head,omitempty"`
-	Options    *Operation `json:"options,omitempty"`
-	Patch      *Operation `json:"patch,omitempty"`
-	Post       *Operation `json:"post,omitempty"`
-	Put        *Operation `json:"put,omitempty"`
-	Parameters Parameters `json:"parameters,omitempty"`
+	Extensions map[string]interface{} `json:"-"`
+	Ref        string                 `json:"$ref,omitempty"`
+	Delete     *Operation             `json:"delete,omitempty"`
+	Get        *Operation             `json:"get,omitempty"`
+	Head       *Operation             `json:"head,omitempty"`
+	Options    *Operation             `json:"options,omitempty"`
+	Patch      *Operation             `json:"patch,omitempty"`
+	Post       *Operation             `json:"post,omitempty"`
+	Put        *Operation             `json:"put,omitempty"`
+	Parameters Parameters             `json:"parameters,omitempty"`
+}
+
+func (pathItem *PathItem) UnmarshalJSON(b []byte) (err error) {
+	type _PathItem PathItem
+	p := _PathItem{}
+	if err := json.Unmarshal([]byte(b), &p); err != nil {
+		return err
+	}
+	if err := json.Unmarshal([]byte(b), &p.Extensions); err != nil {
+		return err
+	}
+	StripNonCustomExtensions(&p.Extensions)
+	*pathItem = PathItem(p)
+	return nil
+}
+
+func (p PathItem) MarshalJSON() ([]byte, error) {
+	return marshallFromMap(p.Extensions, reflect.ValueOf(p), reflect.TypeOf(p))
```

Comment:
Why not use the jsoninfo package that's widely used in openapi3 package?

Reply:
Honestly I was a little intimidated by it at first, but I think I've a better grasp of it now. I'll make that change today 👍